### PR TITLE
[Validator] Add warning for null and blank values for minimum string length

### PR DESCRIPTION
**Pull request by theofidry**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | symfony/symfony#16157 |

Follow up of #5789.

Summary of what has been said: even though the [Length#min](http://symfony.com/doc/current/reference/constraints/Length.html#min) option is well detailed and mention the case of `null` values, the introduction is still confusing:

> Validates that a given string length is between some minimum and maximum value.

As a result, a warning note is being added under this introduction.

I've listed all the Symfony constraints. For some this `null` and `blank` values specificity has already been mentioned. I've noted the ones on which this warning does not apply, but for some I am not sure so it needs to be doubled checked.
- Basic Contraints
  - [x] NotBlank: already done
  - [x] Blank: already done
  - [x] NotNull: already done
  - [x] IsNull: already done
  - ~~[ ] IsTrue~~
  - ~~[ ] IsFalse~~
  - ~~[ ] Type~~
- String Constraints¶
  - ~~[ ] Email~~
  - [x] Length
  - ~~[ ] Url~~
  - ~~[ ] Regex~~
  - ~~[ ] Ip~~
  - ~~[ ] Uuid~~
- Number Constraints¶
  - [ ] Range
- Comparison Constraints¶
  - ~~[ ] EqualTo~~
  - ~~[ ] NotEqualTo~~
  - ~~[ ] IdenticalTo~~
  - ~~[ ] NotIdenticalTo~~
  - [ ] LessThan: needed? How `@Assert\LessThan("-18 years")` behaves if the value is null?
  - [ ] LessThanOrEqual: same as `LessThan`
  - [ ] GreaterThan: same as `LessThan`
  - [ ] GreaterThanOrEqual: same as `LessThan`
- Date Constraints¶
  - ~~[ ] Date~~
  - ~~[ ] DateTime~~
  - ~~[ ] Time~~
- Collection Constraints¶
  - ~~[ ] Choice~~
  - ~~[ ] Collection~~
  - ~~[ ] Count~~
  - ~~[ ] UniqueEntity~~
  - ~~[ ] Language~~
  - ~~[ ] Locale~~
  - ~~[ ] Country~~
- File Constraints¶
  - ~~[ ] File~~
  - ~~[ ] Image~~
- Financial and other Number Constraints¶
  - ~~[ ] Bic~~
  - ~~[ ] CardScheme~~
  - ~~[ ] Currency~~
  - ~~[ ] Luhn~~
  - ~~[ ] Iban~~
  - ~~[ ] Isbn~~
  - ~~[ ] Issn~~
- Other Constraints¶
  - ~~[ ] Callback~~
  - ~~[ ] Expression~~
  - ~~[ ] All~~
  - ~~[ ] UserPassword~~
  - ~~[ ] Valid~~

### CODE REVIEW

File: reference/constraints/Length.rst [hunk 1]
```diff
@@ -1,8 +1,12 @@
 Length
 ======
 
-Validates that a given string length is *between* some minimum and maximum
-value.
+Validates that a given string length is *between* some minimum and maximum value.
+
+**Warning**: *null* and *blank* values are not handled by this constraint. To
+handle them, refers to the :doc:`/reference/constraints/Null`,
```
Comment (xabbuh) [inline reference/constraints/Length.rst]: typo here: refer

Comment (theofidry) [inline reference/constraints/Length.rst]: @xabbuh does this generic warning message is good for you? I would like to agree on the message in one place first before applying it everywhere needed.

Comment (xabbuh) [inline reference/constraints/Length.rst]: I would write it like "`null` and the empty string are not handled by this constraint. Additionally add the `NotBlank` or `NotNull` constraint to exclude these values."

Comment (xabbuh) [inline reference/constraints/Length.rst]: and then we should add it in a `caution` block

**Comment by weaverryan** [comment]
Thanks @theofidry! I've made the final changes. And yes I believe this problem extends also to `LessThan`, `LessThankOrEqual`, etc. I haven't added the notes here - we can do that in another PR.

**Pull request closed without merging**